org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/InvisibleProjectImporter.java

```diff
@@ -105,13 +105,12 @@ public void importToWorkspace(IProgressMonitor monitor) throws OperationCanceled
 				IJavaProject javaProject = JavaCore.create(invisibleProject);
 				ProjectUtils.addSourcePath(sourcePath, subProjectPaths.toArray(new IPath[0]), javaProject);
 				JavaLanguageServerPlugin.logInfo("Successfully created a workspace invisible project " + invisibleProjectName);
+				UpdateClasspathJob.getInstance().updateClasspath(javaProject, preferencesManager.getPreferences().getReferencedLibraries());
```

Comment:
Now we only need to update the classpath when creating a invisible project. After the project is created, its classpath will get updated in following path everytime we reopen it:
1. `InitHandler` calls `preferenceManager.update(pref)` in initialization.
2. `ProgramsManager.prefChangeListener` invokes `UpdateClasspathJob`.